### DISCUSSION TITLE
Move the Vts.Scripting project into a separate repository

### DISCUSSION THREAD

**Discussion by lmalenfant** [VirtualPhotonics/VTS#127] [General]
I would like to move the Vts.Scripting and Vts.Scripting.Test projects into a separate repository that reference the VTS library. The tests for the scripting project take a long time to run because they are not unit tests and for me, this is hindering development.

I rely heavily on the tests and run them frequently and running them for each project individually is not really an option because this is also time consuming.

**Comment by dcuccia** [comment]
Would you be open to considering the tests as integration tests that get run only when a full PR is created, so there's less day-to-day burden on the developer? As a recap, the goal with having the scripts in the same repo was:

a) that these scripts would be tested and compelled alongside the core, so that devs would know if core changes broke the scripts, and 

b) that any additional functionality in the core needed to support new C# scripts, as they were developed, could be added as part of the specific improvement work.

The other way to do a), but in a separate repo would be to trigger automated CI for the scripts repo that happened upon pull of the latest from main on Vts. But that would rely on Vts source, not Nuget, and would be more complex. And it doesn't solve b). This is why I'd tried to get buy-in for the current solution before we started developing it.

**Comment by lmalenfant** [comment]
I did suggest a separate repo when we first started discussing this but that was when were discussing the Python scripts. 

We had the same issue with the WPF but when we make changes to the VTS, we simply pull the WPF project into the solution so we can verify that everything still works as expected. I did this for you if you remember when you made the RangeOfT change. It's very straight-forward and I believe we have documentation on how to do this.

This is the documentation, it needs updating (I will make the necessary changes) because it is a lot easier now:
https://github.com/VirtualPhotonics/VTS/wiki/Including-the-WPF-Application-in-the-Vts-Solution

**Comment by dcuccia** [comment]
Agree the Python should be separate - this was always the plan. These C# scripts are the "bridge" that was designed to help keep a good watch on those. Would suggest any process we come up with is automated. Do you have any thoughts about my question/suggestion?

**Reply by lmalenfant** [reply]
We currently don't have the test projects organized in this way to easily allow for this tiered unit testing.

If it hadn't been hindering my work this week, I would not have made the suggestion but everything takes so much time now. The WPF was causing similar issues which is why we removed that to a separate repository.

**Comment by dcuccia** [comment]
Maybe I can make a quick suggestion on test annotations, so we can opt-into the script testing as appropriate and unblock you. I'd need a couple days as I'm pushing on an internal deadline.

**Reply by lmalenfant** [reply]
Don't worry about this now, let's discuss at our next meeting, I don't need to act on this yet. I can just just unload those projects to improve my productivity.
